awx/main/tasks.py

```diff
@@ -660,6 +660,25 @@ def update_host_smart_inventory_memberships():
         smart_inventory.update_computed_fields()
 
 
+@task()
+def migrate_legacy_event_data(tblname, total_rows):
+    if 'event' not in tblname:
+        return
+    offset = 0
+    chunk = 10000
+    with connection.cursor() as cursor:
+        while offset < total_rows:
+            sql = f'INSERT INTO {tblname} SELECT * FROM _old_{tblname} ORDER BY id DESC OFFSET {offset} LIMIT {chunk};'
```

Comment:
I'm doing this in chunked commits so that:

1.  Users with large amounts of data can see progress (i.e., don't have to wait for one giant INSERT to finish)
2.  If we handle chunked inserts that insert into NEW and delete from OLD in individual commits, and the process is interrupted at some point (i.e., the database restarts), then you can just restart this task by hand.

Reply:
It would be interesting to see the benchmarks without chunking. I'd guess the chunking slows this down, vs. just letting postgre to chunk it. (not sure how much though)

Reply:
I'd show you benchmarks, but I gave up, because the answer is "mind-numbingly slower" :D.  The cost of a transaction per insert if _very_ notable.

Reply:
Based on some feedback from @chrismeyersfsu, I *did* make the "chunk size" configurable so users can lower it if they like.